torchtune/models/llama3/_tokenizer.py

```diff
@@ -268,9 +282,22 @@ def tokenize_messages(
             if self.max_seq_len and len(tokens) >= self.max_seq_len:
                 break
 
+        # Remove final eot if add_final_eot is False
+        # and last message was from the assistant
+        if not add_final_eot:
```

Comment:
Not sure how much I like this...

Reply:
What if there are a bunch of assistant messages in the list? Or what if the assistant message comes before the user message?

Reply:
In what case would you ask for an assistant response where the last message wasn't assistant? Also if there are multiple "prefilled" assistant responses, it would still only care about the last one as that's the one that begins the model's response during decoding.